Cookbook.Infrastructur/Persistance/RecipebookRepository.cs

```diff
@@ -0,0 +1,283 @@
+﻿using Cookbook.Application.Database;
+using Cookbook.Domain.Models;
+using Dapper;
+using System.Data;
+
+namespace Cookbook.Infrastructur.Persistance //restructure DB to use rowID as FK instead of slug
+{
+    public class RecipebookRepository : IRecipebookRepository
+    {
+        private readonly IDbConnectionFactory dbConnectionFactory;
+
+        public RecipebookRepository(IDbConnectionFactory dbConnectionFactory)
+        {
+            this.dbConnectionFactory = dbConnectionFactory;
+        }
+
+        public async Task<bool> CreateAsync(Recipe recipe, CancellationToken token)
+        {
+            using var connection = await dbConnectionFactory.CreateConnectionAsync(token);
```

Comment:
Nem tudom hogy itt  melyik a helyes, minden keresre uj connection-t nyitni, vagy egyet hasznalni per repository. Ennek utana neztel esetleg?

Reply:
konkretan utana nem neztem, videoban meg COnal igy volt. de utananezek majd

Reply:
megmondom oszinten nem irtam meg ilyen kodot szerintem :/ altalaban ami service-eken dolgoztam ez a kod mar megvolt irva, pretty much az elso amit meg kell irnod, meg rohadt ritkan, kb soha nem valtozik. szoval nincs igazan tapasztalatom, nem tudnam megmondani hogy hogy szoktak/szoktuk. 
Ezert mondom hogy ha ezt lattad videoban vagy valahol, akkor biztos jo igy, de ha nem lattad sehol csak igy sikerult, akkor lehet meger egy kis research-ot, hogy hogy sokas. Ha videoban igy volt akkor jo ez igy szerintem.
Using az fontos, de az ott is van

Reply:
a using nem dobja el a vegen mindig a connectiont? es akkor kell uj kovi alkalomra